src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -330,33 +330,37 @@ private static function parseHostPort(array $url, array &$info): array
      */
     private static function dnsResolve(string $host, NativeClientState $multi, array &$info, ?\Closure $onProgress): string
     {
-        if (filter_var(trim($host, '[]'), \FILTER_VALIDATE_IP)) {
-            return $host;
-        }
+        $flag = '' !== $host && '[' === $host[0] && ']' === $host[-1] && str_contains($host, ':') ? \FILTER_FLAG_IPV6 : \FILTER_FLAG_IPV4;
```

Comment:
```suggestion
        $flag = '' !== $host && str_starts_with($host, '[') && str_ends_with($host, ']') && str_contains($host, ':') ? \FILTER_FLAG_IPV6 : \FILTER_FLAG_IPV4;
```

For consistency with `str_contains()`.

Reply:
Sorry, reverted due to:
https://github.com/symfony/symfony/pull/59068#discussion_r1869008859